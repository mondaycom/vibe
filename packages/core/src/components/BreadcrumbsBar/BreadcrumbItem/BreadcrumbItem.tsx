import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import cx from "classnames";
import React, { useRef } from "react";
import useIsOverflowing from "../../../hooks/useIsOverflowing/useIsOverflowing";
import Tooltip from "../../../components/Tooltip/Tooltip";
import { BreadcrumbContent } from "./BreadcrumbContent/BreadcrumbContent";
import { SubIcon, VibeComponentProps } from "../../../types";
import styles from "./BreadcrumbItem.module.scss";

export interface BreadcrumbItemProps extends VibeComponentProps {
  /**
   * The display text of the breadcrumb item.
   */
  text?: string;
  /**
   * If true, the item is disabled.
   */
  disabled?: boolean;
  /**
   * If true, the item is clickable.
   */
  isClickable?: boolean;
  /**
   * The URL the item links to if navigation is handled via a link.
   */
  link?: string;
  /**
   * Callback fired when the item is clicked.
   */
  onClick?: () => void;
  /**
   * If true, applies styling for the current page.
   */
  isCurrent?: boolean;
  /**
   * The icon displayed next to the text.
   */
  icon?: SubIcon;
  /**
   * If true, the breadcrumb text is shown.
   */
  showText?: boolean;
}

const BreadcrumbItem: React.FC<BreadcrumbItemProps> = ({
  className,
  text = "",
  disabled,
  isClickable = false,
  link,
  onClick,
  isCurrent = false,
  icon,
  id,
  showText = true,
  "data-testid": dataTestId
}) => {
  const componentRef = useRef<HTMLSpanElement>(null);
  const isOverflowing = useIsOverflowing({ ref: componentRef });

  return (
    <Tooltip
      disableDialogSlide={true}
      withoutDialog={false}
      content={(isOverflowing || !showText) && text}
      showTrigger={["mouseenter"]}
      hideTrigger={["mouseleave"]}
      addKeyboardHideShowTriggersByDefault={!showText}
    >
      <li
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.BREADCRUMB_ITEM, id)}
        className={cx(styles.breadcrumbItemWrapper, className, {
          [styles.disabled]: disabled
        })}
      >
        <BreadcrumbContent
          ref={componentRef}
          isClickable={isClickable}
          link={link}
          onClick={onClick}
          text={text}
          icon={icon}
          isCurrent={isCurrent}
          disabled={disabled}
          showText={showText}
        />
      </li>
    </Tooltip>
  );
};

export default BreadcrumbItem;
