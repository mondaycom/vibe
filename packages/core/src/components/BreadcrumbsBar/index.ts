export { default as BreadcrumbsBar, BreadcrumbBarProps } from "./BreadcrumbsBar";
export { default as BreadcrumbItem, BreadcrumbItemProps } from "./BreadcrumbItem/BreadcrumbItem";
export { default as BreadcrumbMenu, BreadcrumbMenuProps } from "./BreadcrumbMenu/BreadcrumbMenu";
export {
  default as BreadcrumbMenuItem,
  BreadcrumbMenuItemProps
} from "./BreadcrumbMenu/BreadcrumbMenuItem/BreadcrumbMenuItem";

export * from "./Breadcrumbs.types";
