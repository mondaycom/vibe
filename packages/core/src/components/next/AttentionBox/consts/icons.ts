import { Info, Check, Alert, Warning, Favorite } from "@vibe/icons";
import { AttentionBoxType } from "../AttentionBox.types";
import { SubIcon } from "../../../../types";

export const ATTENTION_BOX_DEFAULT_ICONS: Record<AttentionBoxType, SubIcon> = {
  primary: Info,
  success: Check,
  danger: Alert,
  warning: Warning,
  dark: Favorite
};
