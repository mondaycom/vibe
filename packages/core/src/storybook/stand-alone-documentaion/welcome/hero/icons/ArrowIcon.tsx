import React from "react";

export default function ArrowIcon({ className }: { className?: string }) {
  return (
    <svg
      className={className}
      width="146"
      height="89"
      viewBox="0 0 146 89"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M142.191 52.8681L107.627 41.1405M142.191 52.8681L114.507 85.239M142.191 52.8681C50.1209 94.2659 34.4833 12.7837 63.7408 4.73466C92.9982 -3.31434 96.5293 100.303 4.0957 57.5865"
        stroke="#FF5A85"
        strokeWidth="7.14286"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
}
