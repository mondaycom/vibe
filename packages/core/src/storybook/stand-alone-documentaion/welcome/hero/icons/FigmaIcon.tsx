import React from "react";

export default function FigmaIcon() {
  return (
    <svg width="22" height="22" viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M10.9333 11.2388C10.9333 10.3743 11.2767 9.54534 11.888 8.9341C12.4992 8.32288 13.3282 7.97949 14.1926 7.97949C15.057 7.97949 15.886 8.32288 16.4973 8.9341C17.1085 9.54534 17.4519 10.3743 17.4519 11.2388C17.4519 12.1032 17.1085 12.9322 16.4973 13.5434C15.886 14.1546 15.057 14.498 14.1926 14.498C13.3282 14.498 12.4992 14.1546 11.888 13.5434C11.2767 12.9322 10.9333 12.1032 10.9333 11.2388Z"
        fill="#1ABCFE"
      />
      <path
        d="M4.41455 17.7568C4.41455 16.8924 4.75794 16.0634 5.36917 15.4521C5.98039 14.8409 6.8094 14.4976 7.67381 14.4976H10.9331V17.7568C10.9331 18.6212 10.5897 19.4503 9.97846 20.0615C9.36723 20.6726 8.53822 21.0161 7.67381 21.0161C6.8094 21.0161 5.98039 20.6726 5.36917 20.0615C4.75794 19.4503 4.41455 18.6212 4.41455 17.7568Z"
        fill="#0ACF83"
      />
      <path
        d="M10.9333 1.46045V7.97897H14.1926C15.057 7.97897 15.886 7.63558 16.4973 7.02436C17.1085 6.41313 17.4519 5.58412 17.4519 4.71971C17.4519 3.8553 17.1085 3.02629 16.4973 2.41506C15.886 1.80383 15.057 1.46045 14.1926 1.46045H10.9333Z"
        fill="#FF7262"
      />
      <path
        d="M4.41455 4.71971C4.41455 5.58412 4.75794 6.41313 5.36917 7.02436C5.98039 7.63558 6.8094 7.97897 7.67381 7.97897H10.9331V1.46045H7.67381C6.8094 1.46045 5.98039 1.80383 5.36917 2.41506C4.75794 3.02629 4.41455 3.8553 4.41455 4.71971Z"
        fill="#F24E1E"
      />
      <path
        d="M4.41455 11.2388C4.41455 12.1032 4.75794 12.9322 5.36917 13.5434C5.98039 14.1546 6.8094 14.498 7.67381 14.498H10.9331V7.97949H7.67381C6.8094 7.97949 5.98039 8.32288 5.36917 8.9341C4.75794 9.54534 4.41455 10.3743 4.41455 11.2388Z"
        fill="#A259FF"
      />
    </svg>
  );
}
