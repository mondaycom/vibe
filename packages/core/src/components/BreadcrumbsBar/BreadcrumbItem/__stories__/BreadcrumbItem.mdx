import { Meta } from "@storybook/blocks";
import * as BreadcrumbItemStories from "./BreadcrumbItem.stories";

<Meta of={BreadcrumbItemStories} />

# BreadcrumbItem

- [Overview](#overview)
- [Props](#props)
- [Variants](#variants)
- [Feedback](#feedback)

## Overview

<Canvas of={BreadcrumbItemStories.Overview} />

### Import

```js
import { BreadcrumbItem } from "@vibe/core";
```

## Props

<PropsTable />

## Variants

### States

<Canvas of={BreadcrumbItemStories.States} />

### With icon

<Canvas of={BreadcrumbItemStories.WithIcon} />
