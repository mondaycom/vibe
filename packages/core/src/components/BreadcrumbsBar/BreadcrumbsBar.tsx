import cx from "classnames";
import React, { FC, ReactElement } from "react";
import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
import { NavigationChevronRight } from "@vibe/icons";
import { BreadcrumbsBarType as BreadcrumbsBarTypeEnum } from "./BreadcrumbsConstants";
import { BreadcrumbsBarType } from "./Breadcrumbs.types";
import { BreadcrumbItemProps } from "./BreadcrumbItem/BreadcrumbItem";
import { withStaticProps, VibeComponentProps } from "../../types";
import styles from "./BreadcrumbsBar.module.scss";

export interface BreadcrumbBarProps extends VibeComponentProps {
  /**
   * The type of the breadcrumb bar, determining if it is navigational or for indication only.
   */
  type: BreadcrumbsBarType;
  /**
   * The breadcrumb items displayed in the bar.
   */
  children: ReactElement<BreadcrumbItemProps> | ReactElement<BreadcrumbItemProps>[];
}

const BreadcrumbsBar: FC<BreadcrumbBarProps> & { types?: typeof BreadcrumbsBarTypeEnum } = ({
  className,
  children,
  type = "indication",
  id,
  "data-testid": dataTestId
}: BreadcrumbBarProps) => (
  <nav
    aria-label="Breadcrumb"
    className={cx(styles.breadcrumbsBar, className)}
    id={id}
    data-testid={dataTestId || getTestId(ComponentDefaultTestId.BREADCRUMBS_BAR, id)}
  >
    <ol>
      {children &&
        React.Children.map(children, (child, index) =>
          React.isValidElement(child)
            ? [
                index > 0 && <NavigationChevronRight className={styles.separatorIcon} size="14" aria-hidden="true" />,
                React.cloneElement(child, {
                  ...child?.props,
                  isClickable: type !== "indication"
                })
              ]
            : null
        )}
    </ol>
  </nav>
);

export default withStaticProps(BreadcrumbsBar, {
  types: BreadcrumbsBarTypeEnum
});
