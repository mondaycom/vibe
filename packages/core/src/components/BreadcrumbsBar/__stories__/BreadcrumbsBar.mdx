import { Meta } from "@storybook/blocks";
import BreadcrumbsBar from "../BreadcrumbsBar";
import BreadcrumbItem from "../BreadcrumbItem/BreadcrumbItem.tsx";
import Avatar from "../../Avatar/Avatar";
import DialogContentContainer from "../../DialogContentContainer/DialogContentContainer";
import { Board, Folder, Group, Workspace } from "@vibe/icons";
import person3 from "./assets/person3.png";
import {
  BUTTON_GROUP,
  MULTI_STEP_INDICATOR,
  TABS
} from "../../../storybook/components/related-components/component-description-map";
import { TipCheckYourself } from "./BreadcrumbsBar.stories.helpers";
import * as BreadcrumbsBarStories from "./BreadcrumbsBar.stories";

<Meta of={BreadcrumbsBarStories} />

# BreadcrumbsBar

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Do’s and don’ts](#dos-and-donts)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

Breadcrumbs allow users to keep track and maintain awareness of their location as they navigate through pages, folders, files, etc.

<Canvas of={BreadcrumbsBarStories.Overview} />

### Import

```js
import { BreadcrumbsBar } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Breadcrumbs show users their current location relative to the information architecture and enable them to quickly move up to a parent level or previous step",
    "Effective in a hierarchy of more than two levels.",
    "Start with the highest level parent page and move deeper into the information architecture as the breadcrumb trail progresses.",
    "Use breadcrumbs when the user is most likely to have landed on the page from an external source.",
    "Place breadcrumbs at the top left corner of the screen, above the page title."
  ]}
/>

<TipCheckYourself />

## Variants

### Text only

<Canvas of={BreadcrumbsBarStories.TextOnly} />

### With icons

<Canvas of={BreadcrumbsBarStories.WithIcons} />

### With Breadcrumb Menu

<Canvas of={BreadcrumbsBarStories.WithBreadcrumbMenu} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      positive: {
        component: (
          <BreadcrumbsBar type="navigation">
            <BreadcrumbItem text="Workspace" icon={Workspace} isCurrent />
            <BreadcrumbItem text="Folder" icon={Folder} />
            <BreadcrumbItem text="Board" icon={Board} />
            <BreadcrumbItem text="Group" icon={Group} />
          </BreadcrumbsBar>
        ),
        description: "If there’s a need to insert an icon, use for all items."
      },
      negative: {
        component: (
          <BreadcrumbsBar type="navigation">
            <BreadcrumbItem text="Workspace" icon={Workspace} isCurrent />
            <BreadcrumbItem text="Folder" />
            <BreadcrumbItem text="Board" />
            <BreadcrumbItem text="Group" />
          </BreadcrumbsBar>
        ),
        description: "Don’t use icons if not applied to all steps."
      }
    },
    {
      positive: {
        component: (
          <DialogContentContainer className="monday-storybook-breadcrumbs-bar_inline-wrapper">
            <Avatar size="medium" src={person3} type="img" />
            <div className="monday-storybook-breadcrumbs-bar_column-wrapper">
              <span className="monday-storybook-breadcrumbs-bar_title">Rotem Dekel</span>
              <BreadcrumbsBar type="navigation">
                <BreadcrumbItem text="User research" icon={Board} />
                <BreadcrumbItem text="Video sessions" icon={Group} />
              </BreadcrumbsBar>
            </div>
          </DialogContentContainer>
        ),
        description: "Use breadcrumbs when there is more than two levels of hierarchy."
      },
      negative: {
        component: (
          <DialogContentContainer className="monday-storybook-breadcrumbs-bar_inline-wrapper">
            <Avatar size="medium" src={person3} type="img" />
            <div className="monday-storybook-breadcrumbs-bar_column-wrapper">
              <span className="monday-storybook-breadcrumbs-bar_title">Rotem Dekel</span>
              <BreadcrumbsBar type="navigation">
                <BreadcrumbItem text="User research" icon={Board} />
              </BreadcrumbsBar>
            </div>
          </DialogContentContainer>
        ),
        description: "Don’t use breadcrumbs when there is only one level of navigation or hierachy."
      }
    }
  ]}
/>

## Use cases and examples

### Navigatable breadcrumbs

Use when breadcrumbs are clickable and are used for information and navigation

<Canvas of={BreadcrumbsBarStories.NavigatableBreadcrumbs} />

## Related components

<RelatedComponents componentsNames={[BUTTON_GROUP, MULTI_STEP_INDICATOR, TABS]} />
