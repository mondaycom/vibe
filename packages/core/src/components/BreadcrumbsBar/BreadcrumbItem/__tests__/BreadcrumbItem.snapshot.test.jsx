import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import { Board } from "@vibe/icons";
import BreadcrumbsBar from "../../BreadcrumbsBar";
import BreadcrumbItem from "../BreadcrumbItem";

describe("BreadcrumbsItem renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with icon", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem icon={Board} />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with current item", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem isCurrent />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("when disabled item", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem disabled />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with className", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem className="test" />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with text", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem text="item" />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
