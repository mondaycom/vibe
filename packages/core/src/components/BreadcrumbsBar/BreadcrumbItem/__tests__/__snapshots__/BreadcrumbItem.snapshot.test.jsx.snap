// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BreadcrumbsItem renders correctly > when disabled item 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper disabled"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent disabled notClickable"
        tabIndex={-1}
      >
        <span
          className="breadcrumbText"
        >
          
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`BreadcrumbsItem renders correctly > with className 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper test"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`BreadcrumbsItem renders correctly > with current item 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-current="page"
        aria-disabled="true"
        className="breadcrumbContent current notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`BreadcrumbsItem renders correctly > with empty props 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`BreadcrumbsItem renders correctly > with icon 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent notClickable"
        tabIndex={0}
      >
        <svg
          className="breadcrumbIcon"
          clickable={false}
          fill="currentColor"
          height={14}
          viewBox="0 0 20 20"
          width={14}
        >
          <path
            clipRule="evenodd"
            d="M7.5 4.5H16C16.2761 4.5 16.5 4.72386 16.5 5V15C16.5 15.2761 16.2761 15.5 16 15.5H7.5L7.5 4.5ZM6 4.5H4C3.72386 4.5 3.5 4.72386 3.5 5V15C3.5 15.2761 3.72386 15.5 4 15.5H6L6 4.5ZM2 5C2 3.89543 2.89543 3 4 3H16C17.1046 3 18 3.89543 18 5V15C18 16.1046 17.1046 17 16 17H4C2.89543 17 2 16.1046 2 15V5Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
        <span
          className="breadcrumbText"
        >
          
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`BreadcrumbsItem renders correctly > with text 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          item
        </span>
      </span>
    </li>
  </ol>
</nav>
`;
